shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/common/segment/ddl/routine/ValidStatementSegment.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.sql.parser.sql.common.segment.ddl.routine;
+
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import org.apache.shardingsphere.sql.parser.sql.common.segment.SQLSegment;
+import org.apache.shardingsphere.sql.parser.sql.common.statement.ddl.AlterTableStatement;
+import org.apache.shardingsphere.sql.parser.sql.common.statement.ddl.CreateTableStatement;
+import org.apache.shardingsphere.sql.parser.sql.common.statement.ddl.DropTableStatement;
+import org.apache.shardingsphere.sql.parser.sql.common.statement.ddl.TruncateStatement;
+import org.apache.shardingsphere.sql.parser.sql.common.statement.dml.DeleteStatement;
+import org.apache.shardingsphere.sql.parser.sql.common.statement.dml.InsertStatement;
+import org.apache.shardingsphere.sql.parser.sql.common.statement.dml.SelectStatement;
+import org.apache.shardingsphere.sql.parser.sql.common.statement.dml.UpdateStatement;
+
+import java.util.Optional;
+
+/**
+ * Valid statement segment.
+ */
+@RequiredArgsConstructor
+@Getter
+@Setter
+public class ValidStatementSegment implements SQLSegment {
+    
+    private final int startIndex;
+    
+    private final int stopIndex;
+    
+    private CreateTableStatement createTable;
+    
+    private AlterTableStatement alterTable;
+    
+    private DropTableStatement dropTable;
+    
+    private TruncateStatement truncate;
+    
+    private InsertStatement insert;
+    
+    private InsertStatement replace;
+    
+    private UpdateStatement update;
+    
+    private DeleteStatement delete;
+    
+    private SelectStatement select;
```

Comment:
There only will be one of them at a time, right?  Plus, currently, do we indeed care about the concrete class type of this object?  I mean, do you think it is possible to merge them as a `SQLStatement`?

Reply:
> There only will be one of them at a time, right? Plus, currently, do we indeed care about the concrete class type of this object? I mean, do you think it is possible to merge them as a `SQLStatement`?

@tristaZero For now, we are only concerned with the tables contained in these statements, which are then used for validation, and perhaps `sharding routing and rewriting` will be supported in the future.

If we combine these statements into a single statement, we need to extract the tables contained in the various statements during parsing. This will add a lot of logic to the visitor. Is that ok? 🤔

Reply:
@tristaZero I like your suggestion that using `SQLStatement` to receive different types of statements, and then use public methods to provide external calls. 👍